port/unix/omrsysinfo.c

```diff
@@ -6775,13 +6901,49 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 		 */
 		size_t len = strlen(metricElement->value);
 		metricElement->units = currentElement->metricUnit;
-		if ((len > 0) && (metricElement->value[len-1] == '\n')) {
+		if ((len > 1) && (metricElement->value[len-1] == '\n')) {
 			metricElement->value[len-1] = '\0';
 		}
 		if (currentElement->isValueToBeChecked) {
-			int64_t result = 0;
-			sscanf(metricElement->value, "%" PRId64, &result);
-			if ((result > (MAX_DEFAULT_VALUE_CHECK)) || (result < 0)) {
+			BOOLEAN notSet = FALSE;
+			if ('\n' == metricElement->value[0]) {
+				/* Empty file, see cpuset.cpus and cpuset.mems in omrCgroupCpusetMetricMapV2. */
+				notSet = TRUE;
+			} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+				int64_t result = 0;
+				rc = sscanf(metricElement->value, "%" PRId64, &result);
+				if (1 != rc) {
+					rc = portLibrary->error_set_last_error_with_message_format(
+							portLibrary,
+							OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_FILE_INVALID_VALUE,
+							"invalid value %s in file %s",
+							metricElement->value,
+							subsystemMetricMapElement->metricFileName);
+				} else {
+					if ((result > MAX_DEFAULT_VALUE_CHECK) || (result < 0)) {
+						notSet = TRUE;
+					}
+					rc = 0;
+				}
+			} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE)) {
+				uint64_t uresult = 0;
+				rc = scanCgroupIntOrMax(portLibrary, metricElement->value, &uresult);
+				if (0 != rc) {
+					rc = portLibrary->error_set_last_error_with_message_format(
+							portLibrary,
+							OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_FILE_INVALID_VALUE,
+							"invalid value %s in file %s",
+							metricElement->value,
+							subsystemMetricMapElement->metricFileName);
+				} else if (uresult > MAX_DEFAULT_VALUE_CHECK) {
+					/* Assume that the metric is effectively unlimited and can be considered "Not Set". */
+					notSet = TRUE;
+				}
+			} else {
+				Trc_PRT_Assert_ShouldNeverHappen();
+			}
+			/* Check that the metric is set and that it is a reasonable value. */
+			if (notSet) {
```

Comment:
If `scanCgroupIntOrMax` or `sscanf` error i.e. `(0 != rc)`, `notSet` will currently be set to `FALSE`. Should these error cases also set `notSet` to `TRUE`?

Reply:
I think we should leave it as is, ignore the scan error and output the metric string